test/metric_learn_test.py

```diff
@@ -149,5 +149,48 @@ def test_iris(self):
     self.assertLess(csep, 0.25)
 
 
+class TestPGDM(MetricTestCase):
+  def test_iris(self):
+
+    # Generate full set of constraints for comparison with reference implementation
+    n = self.iris_points.shape[0]
+    a, b, c, d = [], [], [], []
+    for i in range(n):
+      for j in range(i+1, n):
+        if self.iris_labels[i] == self.iris_labels[j]:
+          a.append(i)
+          b.append(j)
+        else:
+          c.append(i)
+          d.append(j)
```

Comment:
This can be made a bit nicer with vectorization:
```python
mask = self.iris_labels[None] == self.iris_labels[:,None]
a, b = np.nonzero(np.triu(mask, k=1))
c, d = np.nonzero(np.triu(~mask, k=1))
```